### PR TITLE
refactor(core): migrate momentjs functionality to luxon + date-fns

### PR DESCRIPTION
**Pull request by anotherchrisberry**
momentjs is being replaced by luxon, so we might as well start moving that way. We'll get some smaller bundles (I think moment + timezones is ~220kb minified; date-fns + luxon is ~60kb minified).

Unfortunately, luxon does not support the `humanize` function, so we need to pull in something for that; date-fns is pretty close to what moment does. It would be nice to _just_ use date-fns; unfortunately, it does not support timezones, so we've got both here. It looks like date-fns is working on timezone support, so that might change. At least date-fns publishes functions on their own, so we're only bringing in what we need?

**Worth considering**
* We're not getting rid of momentjs any time soon. It's a transitive dependency of chart.js (used by kayenta), and we have numerous dependencies in our internal build at Netflix (astrid, tvui-dashboard, react-date-range). So, really, we're just adding to our bundle for now.
* The humanized text is slightly different in places, so if someone is depending on this for functional testing or screen scraping (I hope nobody is screen scraping duration text), they might have problems (beyond the fact that they're scraping that text for something)

### CODE REVIEW

File: app/scripts/modules/core/src/pipeline/filter/executionFilter.service.ts [hunk 1]
```diff
@@ -11,25 +11,44 @@ import { FilterModelService, ISortFilter } from 'core/filterModel';
 import { Registry } from 'core/registry';
 
 const boundaries = [
-  { name: 'Today', after: () => moment().startOf('day') },
```
Comment (erikmunson) [inline app/scripts/modules/core/src/pipeline/filter/executionFilter.service.ts]: is `Today` intentionally omitted here?

Comment (anotherchrisberry) [reply app/scripts/modules/core/src/pipeline/filter/executionFilter.service.ts]: I don't know what you're talking about.

Comment (erikmunson) [reply app/scripts/modules/core/src/pipeline/filter/executionFilter.service.ts]: nothing

**Comment by christopherthielen** [comment]
nice, am happy about this.  Were you able to do any find/replace or did you have to manually update everything?

**Comment by anotherchrisberry** [comment]
@christopherthielen I was able to do some find, but not replace

luxon objects are immutable, so they are a little different to work with. It really wasn't bad.

**Pull request merged**